http://tuxpaint.cvs.sourceforge.net/viewvc/tuxpaint/tuxpaint/src/tuxpaint.c?r1=1.828&r2=1.829
No longer directly accessing PNG structure members directly, now using png_set_IHDR(). (Patch from SF.net Bug #3386433)

--- a/src/tuxpaint.c	2011/07/29 21:27:39	1.828
+++ b/src/tuxpaint.c	2011/08/08 17:59:12	1.829
@@ -13425,12 +13425,7 @@
       {
 	png_init_io(png_ptr, fi);
 
-	info_ptr->width = surf->w;
-	info_ptr->height = surf->h;
-	info_ptr->bit_depth = 8;
-	info_ptr->color_type = PNG_COLOR_TYPE_RGB;
-	info_ptr->interlace_type = 1;
-	info_ptr->valid = 0;	/* will be updated by various png_set_FOO() functions */
+        png_set_IHDR(png_ptr, info_ptr, surf->w, surf->h, 8, PNG_COLOR_TYPE_RGB, 1, PNG_COMPRESSION_TYPE_BASE, PNG_FILTER_TYPE_BASE);
 
 	png_set_sRGB_gAMA_and_cHRM(png_ptr, info_ptr,
 				   PNG_sRGB_INTENT_PERCEPTUAL);
